dll/win32/shell32/wine/control.c

```diff
@@ -938,94 +1037,111 @@ static	void	Control_DoLaunch(CPanel* panel, HWND hWnd, LPCWSTR wszCmd)
     if ((extraPmts) && (*extraPmts == '@') && (sp == -1)) {
         sp = atoiW(extraPmts + 1);
     }
+#endif
 
     TRACE("cmd %s, extra %s, sp %d\n", debugstr_w(buffer), debugstr_w(extraPmts), sp);
 
     applet = Control_LoadApplet(hWnd, buffer, panel);
     if (applet)
     {
 #ifdef __REACTOS__
-    ULONG_PTR cookie;
-    BOOL bActivated;
-    ATOM aCPLName;
-    ATOM aCPLFlags;
-    ATOM aCPLPath;
-    AppDlgFindData findData;
+        ULONG_PTR cookie;
+        BOOL bActivated;
+        ATOM aCPLName;
+        ATOM aCPLFlags;
+        ATOM aCPLPath;
+        AppDlgFindData findData;
+        
+        if (sp == -1 && buffer2[0] == L'\0')
+        {
+            sp = 0;
+        }
 #endif
+
         /* we've been given a textual parameter (or none at all) */
         if (sp == -1) {
             while ((++sp) != applet->count) {
                 TRACE("sp %d, name %s\n", sp, debugstr_w(applet->info[sp].name));
 
+#ifdef __REACTOS__
+                if (StrCmpIW(buffer2, applet->info[sp].name) == 0)
+#else
                 if (StrCmpIW(extraPmts, applet->info[sp].name) == 0)
+#endif
                     break;
             }
         }
 
-        if (sp >= applet->count) {
-            WARN("Out of bounds (%u >= %u), setting to 0\n", sp, applet->count);
-            sp = 0;
-        }
-
 #ifdef __REACTOS__
         bActivated = (applet->hActCtx != INVALID_HANDLE_VALUE ? ActivateActCtx(applet->hActCtx, &cookie) : FALSE);
 
-        aCPLPath = GlobalFindAtomW(applet->cmd);
-        if (!aCPLPath)
+        if (sp < applet->count)
         {
-            aCPLPath = GlobalAddAtomW(applet->cmd);
-        }
+            aCPLPath = GlobalFindAtomW(applet->cmd);
+            if (!aCPLPath)
+            {
+                aCPLPath = GlobalAddAtomW(applet->cmd);
+            }
 
-        aCPLName = GlobalFindAtomW(L"CPLName");
-        if (!aCPLName)
-        {
-            aCPLName = GlobalAddAtomW(L"CPLName");
-        }
+            aCPLName = GlobalFindAtomW(L"CPLName");
```

Comment:
Please refrain from reformatting code

Reply:
I don't understand. I've surrounded that code with an if, why shouldn't the indent level change? I have made some other refactoring changes to which you might be referring, but I don't know which ones. (Not arguing, just making sure I understand what to do).

Reply:
Less changes, less efforts to understand.

Reply:
I have reverted the formatting changes. However, I still don't understand why that if shouldn't change the indentation.